client/web/src/enterprise/code-monitoring/CreateCodeMonitorPage.tsx

```
@@ -56,17 +56,29 @@ const AuthenticatedCreateCodeMonitorPage: React.FunctionComponent<CreateCodeMoni
                 },
                 trigger: { query: codeMonitor.trigger.query },
 
-                actions: codeMonitor.actions.nodes.map(action => ({
-                    email:
-                        action.__typename === 'MonitorEmail'
-                            ? {
-                                  enabled: action.enabled,
-                                  priority: MonitorEmailPriority.NORMAL,
-                                  recipients: [authenticatedUser.id],
-                                  header: '',
-                              }
-                            : undefined,
-                })),
+                actions: codeMonitor.actions.nodes.map(action => {
+                    switch (action.__typename) {
+                        case 'MonitorEmail':
+                            return {
+                                email: {
+                                    enabled: action.enabled,
+                                    priority: MonitorEmailPriority.NORMAL,
+                                    recipients: [authenticatedUser.id],
+                                    header: '',
+                                },
+                            }
+                        case 'MonitorSlackWebhook':
+                            return {
+                                slackWebhook: {
+                                    enabled: action.enabled,
+                                    url: action.url,
+                                },
+                            }
+                        default:
+                            console.warn('Unknown monitor action type', action.__typename)
```

Maybe we can just throw an error here?

Or, you could also pre-filter the nodes array using something like (I'm not sure this works): `nodes.filter((action): action is IMonitorEmail | IMonitorSlackWebhook => action.__typename === 'MonitorEmail' || action.__typename === 'MonitorSlackWebhook').map(...)`

Same comment for `client/web/src/enterprise/code-monitoring/ManageCodeMonitorPage.tsx`.

---

I like prefiltering better. I didn't want to throw an error in case someone is using the GraphQL API to create actions we don't have UI for yet.

---

Based on this, I added some better conversion logic here in a new file, `actions-converters.ts`. Would appreciate it if you took a second look. :)